engine/gamesys/src/gamesys/components/box2d/comp_collision_object_box2d.cpp

```diff
@@ -0,0 +1,1349 @@
+            return false;
+        }
+
+        dmPhysics::HCollisionShape2D shape2d = dmPhysics::GetCollisionShape2D(world->m_World2D, component->m_Object2D, shape_ix);
+
+        switch(shape_info->m_Type)
+        {
+            case dmPhysicsDDF::CollisionShape::TYPE_SPHERE:
+            {
+                dmPhysics::SetCollisionShapeRadius2D(world->m_World2D, shape2d, shape_info->m_SphereDiameter * 0.5f);
+                dmPhysics::SynchronizeObject2D(world->m_World2D, component->m_Object2D);
+            } break;
+            case dmPhysicsDDF::CollisionShape::TYPE_BOX:
+            {
+                dmPhysics::SetCollisionShapeBoxDimensions2D(world->m_World2D, shape2d, _component->m_Resource->m_ShapeRotation[shape_ix], shape_info->m_BoxDimensions[0] * 0.5f, shape_info->m_BoxDimensions[1] * 0.5f);
+            } break;
+            default: assert(0);
+        }
+
+        return true;
+    }
+
+    static PhysicsEngineType GetPhysicsEngineTypeBox2D(CollisionWorld* _world)
+    {
+        return PhysicsEngineType::PHYSICS_ENGINE_BOX2D;
+    }
+
+    static void InstallBox2DPhysicsAdapter()
+    {
+        if (g_PhysicsAdapter)
+        {
+            return;
+        }
+
+        g_PhysicsAdapter = new PhysicsAdapterFunctionTable();
+        g_PhysicsAdapter->m_IsEnabled              = IsEnabledBox2D;
+        g_PhysicsAdapter->m_WakeupCollision        = WakeupCollisionBox2D;
+        g_PhysicsAdapter->m_RayCast                = RayCastBox2D;
+        g_PhysicsAdapter->m_SetGravity             = SetGravityBox2D;
+        g_PhysicsAdapter->m_GetGravity             = GetGravityBox2D;
+        g_PhysicsAdapter->m_SetCollisionFlipH      = SetCollisionFlipHBox2D;
+        g_PhysicsAdapter->m_SetCollisionFlipV      = SetCollisionFlipVBox2D;
+        g_PhysicsAdapter->m_GetCollisionGroup      = GetCollisionGroupBox2D;
+        g_PhysicsAdapter->m_SetCollisionGroup      = SetCollisionGroupBox2D;
+        g_PhysicsAdapter->m_GetCollisionMaskBit    = GetCollisionMaskBitBox2D;
+        g_PhysicsAdapter->m_SetCollisionMaskBit    = SetCollisionMaskBitBox2D;
+        g_PhysicsAdapter->m_UpdateMass             = UpdateMassBox2D;
+        g_PhysicsAdapter->m_GetShapeIndex          = GetShapeIndexShared;
+        g_PhysicsAdapter->m_GetShape               = GetShapeBox2D;
+        g_PhysicsAdapter->m_SetShape               = SetShapeBox2D;
+        g_PhysicsAdapter->m_CreateJoint            = CreateJointBox2D;
+        g_PhysicsAdapter->m_DestroyJoint           = DestroyJointBox2D;
+        g_PhysicsAdapter->m_GetJointParams         = GetJointParamsBox2D;
+        g_PhysicsAdapter->m_GetJointType           = GetJointTypeBox2D;
+        g_PhysicsAdapter->m_SetJointParams         = SetJointParamsBox2D;
+        g_PhysicsAdapter->m_GetJointReactionForce  = GetJointReactionForceBox2D;
+        g_PhysicsAdapter->m_GetJointReactionTorque = GetJointReactionTorqueBox2D;
+        g_PhysicsAdapter->m_GetPhysicsEngineType   = GetPhysicsEngineTypeBox2D;
+    }
+}
```

Comment:
Next part in the saga may be to fully move the collision objects to an extension as well, so it's not initialized by gamesys.cpp but with a static constructor here.
But that's for an upcoming PR.